framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileElementTypes.java

```diff
@@ -313,6 +321,82 @@ public AnnotatedTypeMirror getAnnotatedTypeMirror(Element e) {
         return Collections.emptySet();
     }
 
+    /**
+     * Returns the method type of the most specific fake override for the given element, when used
+     * as a member of the given type.
+     *
+     * @param elt element for which annotations are returned
+     * @param receiverType the type of the class that contains member (or a subtype of it)
+     * @return the most specific AnnotatedTypeMirror for {@code elt} that is a fake override, or
+     *     null if there are no fake overrides
+     */
+    public @Nullable AnnotatedTypeMirror getFakeOverride(
+            Element elt, AnnotatedTypeMirror receiverType) {
+        if (parsing) {
+            throw new BugInCF("parsing while calling getFakeOverride");
+        }
+
+        if (elt.getKind() != ElementKind.METHOD) {
+            return null;
+        }
+
+        ExecutableElement method = (ExecutableElement) elt;
+
+        TypeMirror methodReceiverType = method.getReceiverType();
+        if (methodReceiverType != null && methodReceiverType.getKind() == TypeKind.NONE) {
+            return null;
+        }
+
+        // This is a list of pairs of (where defined, method type) for fake overrides.  The second
+        // element of each pair is currently always an AnnotatedExecutableType.
+        List<Pair<TypeMirror, AnnotatedTypeMirror>> candidates =
+                annotationFileAnnos.fakeOverrides.get(elt);
+
+        if (candidates == null || candidates.isEmpty()) {
+            return null;
+        }
+
+        TypeMirror receiverTypeMirror = receiverType.getUnderlyingType();
+
+        // A list of fake receiver types.
+        List<TypeMirror> applicable = new ArrayList<>();
+        for (Pair<TypeMirror, AnnotatedTypeMirror> candidatePair : candidates) {
+            TypeMirror fakeLocation = candidatePair.first;
+            AnnotatedExecutableType candidate = (AnnotatedExecutableType) candidatePair.second;
+            if (factory.types.isSameType(receiverTypeMirror, fakeLocation)) {
+                return candidate;
+            } else if (factory.types.isSubtype(receiverTypeMirror, fakeLocation)) {
+                applicable.add(fakeLocation);
+            }
+        }
+
+        if (applicable.isEmpty()) {
+            return null;
+        }
+        TypeMirror fakeReceiverType =
+                TypesUtils.mostSpecific(applicable, factory.getProcessingEnv());
+        if (fakeReceiverType == null) {
+            StringJoiner message = new StringJoiner(System.lineSeparator());
+            message.add(
+                    String.format(
+                            "No most specific fake override found for %s with receiver %s.  These fake overrides are applicable:",
```

Comment:
I've come up with an example that triggers this Exception.  I added it as a test case in `/Users/smillst/jsr308/checker-framework/checker/jtreg/stubs/fakeoverrides`.
```java
interface SuperInterface {
    default int m() {  return 0; }
}

class SuperClass implements SuperInterface {
    // fake override:
    // int m();
}

interface SubInterface extends SuperInterface {
    // fake override:
    // int m();
}
public class Use extends SuperClass implements SubInterface {
    void use(Use d) {
        d.m();  // crash here.
    }
}
```

I don't think this code should crash.  The fake override in `SuperClass` should be used and the fake override in `SubInterface` should be ignored.  In the future, we should issue an error at the declaration of `Use`.